basics.sh

```diff
@@ -1,6 +1,17 @@
 #!/usr/bin/env sh
+
+# NB: All instances of echo have been replaced with printf, due to echo
+# having inconsistent behaviour between different shells. echo was used for
+# printing messages to the shell, and inserting code into the .zshrc file, but
+# does not escape newline characters correctly, which causes messages to be
+# displayed with \n literally printed, and for \n to literally inserted into
+# the .zsh file, instead of being escaped as expected.
+# Therefore, printf is used instead, which behaves as we want, escaping \n
+# characters properly.
+
```

Comment:
Sorry Chris, don't think we actually need this... the person seeing this in the future won't necessarily know that we used to use echo. Please remove but do feel free to add this into the PR description if you don't want to lose it.